ogr/ogrsf_frmts/hana/ogrhanalayer.cpp

```diff
@@ -816,9 +844,31 @@ void OGRHanaLayer::ReadGeometryExtent(int geomField, OGREnvelope *extent)
             extent->MinY = *rsExtent->getDouble(2);
             extent->MaxX = *rsExtent->getDouble(3);
             extent->MaxY = *rsExtent->getDouble(4);
+            set = true;
```

Comment:
Not sure about this. NULL is a valid return for the extent, so it can happen that the method is called again without necessity. On the other hand, this is only the case if the column has no fixed srs, there are no valid geometries or there is unloaded data. The second case shouldn't be much overhead (and seldom) and the third needs a new computation anyway. Is the first one an option in GDAL?

Reply:
As far as I can see, the set variable is only used when fast estimation is applied. And since the M_ST_GEOMETRY_COLUMNS can be empty when data is still in delta, we need to try one more time in line 854.

Reply:
We can load the data in advance. The extent is serialized so it should be cheaper. The only remaining case for NULL values are mixed srids and empty (or only invalid geometries) columns. For the second the other method also yields NULL, do you know what the first case should return for gdal?

Reply:
Hmm I think I don't quite get the point of discussion here. If the fast extent somehow fails, i.e.,

- there is no such row in the table
- there is a row for the column, but the entries are NULL (e.g when the table is empty)
- or some error occurs

we try again with the "standard" way of computing the extent.
I don't think it is important to add logic which tries to interpret the result of the fast extent, i.e., why it failed/return NULL values.

Reply:
Right, no additional logic is needed.

The fast extent estimation may fail if table is unloaded or have geometries with different srids.

Reply:
> We can load the data in advance. The extent is serialized so it should be cheaper. The only remaining case for NULL values are mixed srids and empty (or only invalid geometries) columns. For the second the other method also yields NULL, do you know what the first case should return for gdal?

It is not that simple to load a table, there are some requirements that must be fulfilled, see https://help.sap.com/docs/SAP_HANA_PLATFORM/6b94445c94ae495c83a19646e7c3fd56/c133165bbb57101493c5fb19b5b8607f.html.

```
Prerequisites
You have one of the following privileges:
System privilege TABLE ADMIN
SQL object privilege UPDATE for the table or the schema in which the table is located
```

Reply:
I mean we _could_ probably check the privileges of the current user via. `SELECT * FROM "PUBLIC"."EFFECTIVE_PRIVILEGES" where USER_NAME = CURRENT_USER AND privilege = 'TABLE ADMIN';`.
If there is a record in the result, we can execute the LOAD statement or just execute the statement and ignore the error.
But I guess you are right, this might need further testing and is probably not worth the effort.